server/statistics/hot_peer_cache.go

```diff
@@ -145,21 +142,29 @@ func (f *hotPeerCache) collectRegionMetrics(loads []float64, interval uint64) {
 	}
 }
 
+// CollectExpiredItems collects expired items, mark them as needDelete and puts them into inherit items
+func (f *hotPeerCache) CollectExpiredItems(region *core.RegionInfo) []*HotPeerStat {
+	regionID := region.GetID()
+	items := make([]*HotPeerStat, 0)
+	for _, storeID := range f.getAllStoreIDs(region, true, true) {
+		if region.GetStorePeer(storeID) == nil {
+			item := f.getOldHotPeerStat(regionID, storeID)
+			if item != nil {
+				f.putInheritItem(item)
+				item.needDelete = true
+				items = append(items, item)
+			}
+		}
+	}
+	return items
+}
+
 // CheckRegionFlow checks the flow information of region.
-func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo) (ret []*HotPeerStat) {
+// it is only used for test.
```

Comment:
Why?

Reply:
Only mockcluster used it for test now.

Reply:
CheckReadLeaderSync also uses it?

Reply:
It seems to be better to remove `Check...Sync`.

Reply:
I think we can't remove `Check...Sync`, `Check....Async` isn't friendly enough to be used in unit-test.

Reply:
`CheckRegionFlow` now is removed and the related logic is ported in `mockCluster`.